lib/converter.rb

```diff
@@ -0,0 +1,56 @@
+require 'beta_code'
+
+class Converter
+  # Beta Code specification: https://web.archive.org/web/20140221210442/http://www.tlg.uci.edu/encoding/BCM2013.pdf
+  MORPHEUS_CUSTOM_BETA_TO_GREEK = {
+    # The Beta Code specification says that _ should map to the EM dash
+    # However Morpheus uses _ to represent a macron in <stem> and <suff> tags
+    # The following mapping covers every case that should appear with a macron
+    a_: 'ᾱ',
+    'a)_': 'ἀ̄',
+    'a(_': 'ἁ̄',
+    '*)a_': 'Ἀ̄',
+    '*(a_': 'Ἁ̄',
+    i_: 'ῑ',
+    'i)_': 'ἰ̄',
+    'i(_': 'ἱ̄',
+    '*)i_': 'Ἰ̄',
+    '*(i_': 'Ἱ̄',
+    u_: 'ῡ',
+    'u)_': 'ὐ̄',
+    'u(_': 'ὑ̄',
+    '*(u_': 'Ὑ̄', # there is no *)u_ in Unicode
+
+    # The Beta Code specification uses a number after s to specify different versions of sigma
+    # s is σ or ς, s1 is σ, s2 is ς, s3 is ϲ (lunate sigma), *S is Σ, and *S3 is Ϲ
+    # Morpheus sometimes uses a number at the end of a word to differentiate identical headwords
+    # eg ὅς (rel. pron.) and ὅς2 (poss. pron.)
+    s1: 'ς1',
+    s2: 'ς2',
+    s3: 'ς3',
+    '*s3': 'Σ3',
+
+    # The Beta Code specification says that : should be converted to ·
+    # Morpheus uses : to separate preverbs and augments from the rest of the stem
+    ':': ':',
+  }
+
+  # For the reverse mapping simply remove the macron
+  MORPHEUS_CUSTOM_GREEK_TO_BETA = {
+    _: '',
+  }
+
+  def self.beta_code_to_greek(word)
+    BetaCode.beta_code_to_greek(word, custom_map: MORPHEUS_CUSTOM_BETA_TO_GREEK)
+  end
+
+  def self.greek_to_beta_code(word)
+    BetaCode.greek_to_beta_code(word, custom_map: MORPHEUS_CUSTOM_GREEK_TO_BETA)
+  end
+
+  # To differentiate between identical headwords in Latin, Morpheus appends #1, #2, etc.
+  # In order to match the Greek output we want to remove the #
+  def self.latin_headword(word)
```

Comment:
morpheus uses the headword numbers in Greek too.  E.g. see χράω . These numbers are linkages between the headword and its entry in the lexicons from which morpheus data was derived.  If there were actually URIs for distinct lemmas, then the could be used instead of this system which is not good at all. 

Absent that, I have not known what the right solution is to dealing with the numbers on the morpheus headwords and have waffled back and forth between removing them and keeping them.  

As this impacts the lemma that is stored in the treebank files, it has also impacted those data sets. The lemmas appear in there with and without digits I think, but mostly with.

Reply:
The particular issue this line is trying to solve is that the raw output from Morpheus I tested uses `#2` for Latin but just `2` for Greek.

For example, Latin `qui` returns:

```xml
<dict>
  <hdwd xml:lang="lat">qui#2</hdwd>
  <pofs order="5">pronoun</pofs>
</dict>
```

Greek χράω:

```xml
<dict>
  <hdwd xml:lang="grc-x-beta">xra/w2</hdwd>
  <pofs order="1">verb</pofs>
</dict>
```

I assumed this only applied to Latin, but that assumption is just based on testing a number of words. Do you know if there would be any `#2` added to a Greek word or if there's a better way to figure out if there would be?

Reply:
ah, I see what you mean, sorry I missed that distinction. I don't recall ever seeing the # before the number in the greek lemmas. And they aren't there in the LSJ entry keys so probably not.

Reply:
I've updated the comments to be more clear.